src/components/EmailAuth.vue

```diff
@@ -289,7 +280,40 @@ export default {
     },
     backToSocial () {
       this.$emit('update:emailPass', false)
+    },
+    switchLogin () {
+      this.login = !this.login
+      this.resetPassword = false
+    },
+    sendPasswordResetEmail () {
```

Comment:
There is a small issue with linking account, if you have some lined login method (password provider and social provider), after reseting password, we've got the same error/flow as while linking different providers for the first time. 
Did you try it?
I found one more case - if you register with social provider then, reset password for email (connected to this social), and after updating password, you login with social provider, you have to link it again. 
Please check if there is an option to link those accounts: https://firebase.google.com/docs/auth/web/account-linking?authuser=0
IMO, the password confirmation for social provider is acceptable if needed, but maybe there is a chance to get the state (after reset pass or something) and at least change text/messages in the flow. 
(https://firebase.google.com/docs/auth/web/passing-state-in-email-actions)